subsys/bluetooth/mesh/solicitation.c

```diff
@@ -65,7 +65,7 @@ static int srpl_entry_save(struct bt_mesh_subnet *sub, uint32_t sseq, uint16_t s
 
 	entry = srpl_find_by_addr(ssrc);
 	if (entry) {
-		if (entry->sseq >= sseq && sseq != 0) {
+		if (entry->sseq >= sseq || (entry->sseq == 0 && sseq != 0)) {
```

Comment:
Will it reject all frames except the first one?
1. Device received solicitation frame with sseq == 0.
2. Went to `else` branch, allocated new entry and stored it.
3. Device received next solicitation frame with sseq == 1.
4. This condition is accepted (entry->sseq == 0 && sseq != 0). sseq equal to 0 has already been stored.
5. Device rejects storing sseq == 1
6. Perform reset
7. After reset, perform successful replay attack.

Reply:
And yes, this was obviously a very poor solution!